coralproject/talk
# feat/CORL-2233: do not tag user as staff if outside of their scope

## What does this PR do?
This PR addresses https://vmproduct.atlassian.net/browse/CORL-2233 by treating moderators as normal users when they are commenting on sites outside of their moderation scope.

## What changes to the GraphQL/Database Schema does this PR introduce?
None.

## How do I test this PR?
1. Enable the SITE_MODERATOR feature flag for your talk instance
2. Ensure that at least two sites are added
3. Make a user a Site Moderator on one site
4. As that user, comment on the site for which they are a Site Moderator, observe that the comment is given the STAFF badge, and is auto-approved by "system"
5. As the same user, comment on a site for which they are not a moderator, observe that they do not receive the STAFF badge, and are not auto approved by the system
 
## How do we deploy this PR?
There shouldn't be any special considerations for deploying this PR
